addons/addon-internal-auth-dep-check/packages/internal-auth-dep-check/lib/steps/internal-auth-dep-check-service.js

```diff
@@ -0,0 +1,255 @@
+    const result = await this.dynamoDB
+      .scanner()
+      .table(table)
+      .names({ '#a': 'authenticationProviderId' })
+      .values({ ':i': 'internal' })
+      .filter('#a = :i')
+      .projection(['uid', 'status'])
+      .scan();
+
+    const listOfInternalUsers = _.uniq(result.map(item => item.uid));
+    const listOfInternalStatuses = _.uniq(result.map(item => item.status));
+
+    return { listOfInternalUsers, listOfInternalStatuses };
+  }
+
+  async verifyNoActiveInternalUsers(listOfInternalStatuses) {
+    if (listOfInternalStatuses.includes('active')) {
+      this.log.error(`Found an least one active internal user`);
+      return false;
+    }
+    return true;
+  }
+
+  async verifyNoInternalUserProjects(listOfInternalUsers) {
+    const table = this.settings.get(settingKeys.projectsTableName);
+
+    // Scan for project admins
+    const result = await this.dynamoDB
+      .scanner()
+      .table(table)
+      .projection('projectAdmins')
+      .scan();
+
+    const listOfProjectAdmins = _.uniq(
+      result
+        .map(item => item.projectAdmins)
+        .flat()
+        .filter(item => listOfInternalUsers.includes(item)),
+    );
+
+    if (listOfProjectAdmins.length > 0) {
+      this.log.error(`Found at least one user associated with a project: ${listOfProjectAdmins}`);
+      return false;
+    }
+
+    return true;
+  }
+
+  async verifyNoInternalUserOrgStudies(listOfInternalUsers) {
+    const listOfOrganizationStudyIds = await this._listOrganizationStudyIds();
+    for (let i = 0; i < listOfOrganizationStudyIds.length; i += 1) {
+      // eslint-disable-next-line no-await-in-loop
+      const listOfOrganizationStudyUsers = await this._getPermissionsForStudy(listOfOrganizationStudyIds[i]);
+      for (let j = 0; j < listOfOrganizationStudyUsers.length; j += 1) {
+        if (listOfInternalUsers.includes(listOfOrganizationStudyUsers[j])) {
+          this.log.error(
+            `Found Org Study with internal user permissions: user ${listOfOrganizationStudyUsers[j]} associated with study ${listOfOrganizationStudyIds[i]}`,
+          );
+          return false;
+        }
+      }
+    }
+    for (let i = 0; i < listOfInternalUsers.length; i += 1) {
+      // eslint-disable-next-line no-await-in-loop
+      const listOfInternalUsersStudies = await this._getPermissionsForUser(listOfInternalUsers[i]);
+      for (let j = 0; j < listOfInternalUsersStudies.length; j += 1) {
+        if (listOfOrganizationStudyIds.includes(listOfInternalUsersStudies[j])) {
+          this.log.error(
+            `Found Org Study with internal user permissions: user ${listOfInternalUsers[i]} associated with study ${listOfInternalUsersStudies[j]}`,
+          );
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  async _listOrganizationStudyIds() {
+    const table = this.settings.get(settingKeys.studiesTableName);
+
+    // Scan for Organization studies
+    const result = await this.dynamoDB
```

Comment:
We'll also need to let the deployer know which study-admin to contact for making changes here, if they don't have admin access for this study.